### ISSUE TITLE
Refreshing a page with user script increases the count of user scripts in the exensions tool icon

### ISSUE THREAD

**Issue by Vampire** [Tampermonkey/tampermonkey#1765]
## Expected Behavior

The toolbar icon right of the address bar shows the amount of user scripts that are applied to the current page.

## Actual Behavior

While this is not happening with all user scripts on all pages, it happens 100% reproducibly on https://discuss.gradle.org/latest with the below user script.
When reloading the page, the counter increases by one, so currently it shows "12" instead of "1".

## Specifications

- Chrome: 111.0.5563.147
- TM: 4.18.1
- OS: Windows 10

## Script

```js
// ==UserScript==
// @name          Restore visited topic view on Gradle forums
// @description	  Restores the visited topic view on Gradle forums
// @author        Björn Kautler
// @version       1.2
// @match         https://discuss.gradle.org/*
// @require       https://code.jquery.com/jquery-3.6.3.slim.min.js
// ==/UserScript==
(function() {var css = [
  ".topic-list-item.visited .topic-list-data a.title:not(.badge-notification), .latest-topic-list-item.visited a.title:not(.badge-notification), .category-topic-link.visited a.title:not(.badge-notification) {",
	"    color: var(--primary-medium) !important;",
	"}",
].join("\n");
if (typeof GM_addStyle != "undefined") {
	GM_addStyle(css);
} else if (typeof PRO_addStyle != "undefined") {
	PRO_addStyle(css);
} else if (typeof addStyle != "undefined") {
	addStyle(css);
} else {
	var node = document.createElement("style");
	node.type = "text/css";
	node.appendChild(document.createTextNode(css));
	var heads = document.getElementsByTagName("head");
	if (heads.length > 0) {
		heads[0].appendChild(node);
  }
}

$("body").append('<a id="toggle" href="#" style="position: fixed; left: 10px; top: 10px; z-index: 2000">Toggle visited topics</a>')
$("#toggle").click(function() {
  $(".topic-list-item.visited").toggle("fast")
  return false
})
})();
```

**Comment by Vampire** [comment]
Ah, when changing the "Icon-Information" setting from "Active Userscript Instances" to "Active Userscripts", the displayed value makes more sense.

Also if you have a userscript that works on any URL, and open https://www.tampermonkey.net/index.php with "Active Userscript Instances", the count almost immediately goes up to more than 40 and also increases over time, for example each time a new advertisement is displayed it goes up to 20 more. :-/

**Comment by derjanb** [comment]
Fun Fact: it works once the devtools at https://discuss.gradle.org/latest are open when the page is reloaded. 🤷

**Comment by Vampire** [comment]
I don't see a changed behaviour when devtools are open

**Comment by derjanb** [comment]
This should be fixed at 4.19.6182 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by derjanb** [comment]
Thanks for reporting!
